impeller/geometry/scalar.h

```diff
@@ -52,4 +54,10 @@ struct Degrees {
   };
 };
 
+/// @brief Convert a scalar to a half precision float.
+///
+/// Adapted from
+/// https://developer.android.com/games/optimize/vertex-data-management .
```

Comment:
We should document the behavior of this and how it handles precision loss, and add some tests that assert it.

Specifically:

- What happens when the `f` is greater than the max finite value or less than the lowest finite value of a half float?
- What happens when `f` is infinity/-infinity/nan?
- What happens to very tiny values of `f`?